tuf/repository_lib.py

```diff
@@ -1530,18 +1583,27 @@ def generate_timestamp_metadata(snapshot_filename, version, expiration_date,
   # This check ensures arguments have the appropriate number of objects and
   # object types, and that all dict keys are properly named.
   # Raise 'securesystemslib.exceptions.FormatError' if the check fails.
-  securesystemslib.formats.PATH_SCHEMA.check_match(snapshot_filename)
+  securesystemslib.formats.PATH_SCHEMA.check_match(snapshot_file_path)
   tuf.formats.METADATAVERSION_SCHEMA.check_match(version)
   securesystemslib.formats.ISO8601_DATETIME_SCHEMA.check_match(expiration_date)
   securesystemslib.formats.NAME_SCHEMA.check_match(repository_name)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_length)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_hashes)
 
-  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_fileinfo = {}
-  length, hashes = securesystemslib.util.get_file_details(snapshot_filename,
+
+  length, hashes = securesystemslib.util.get_file_details(snapshot_file_path,
       tuf.settings.FILE_HASH_ALGORITHMS, storage_backend)
+
+  length = (use_length and length) or None
+  hashes = (use_hashes and hashes) or None
+
+  _, snapshot_filename = os.path.split(snapshot_file_path)
```

Comment:
Nit: [`os.path.basename()`](https://docs.python.org/3/library/os.path.html#os.path.basename) would be a bit cleaner here